### PR TITLE
New Scenarios for the System integration test

### PR DESCRIPTION
**Pull request by IgorGuz2000**
## What does this PR do?
This PR testing the System integration by enabling each integration one in a time

## Why is it important?
 most system integration tests are not checking specific integrations that they are running and returning data

## Checklist

- [x] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have run the Unit tests (`make unit-test`), and they are passing locally
- [x] I have run the End-2-End tests for the suite I'm working on, and they are passing locally
- [ ] I have noticed new Go dependencies (run `make notice` in the proper directory)

## Author's Checklist

- [ ]

## How to test this PR locally
OP_LOG_LEVEL=DEBUG  DEV=true godog -t '@system_integration'

## Related issues

-

### CODE REVIEW

File: e2e/_suites/fleet/features/system_integration.feature
Comment (mdelapenya) [inline e2e/_suites/fleet/features/system_integration.feature:2]: There is a preference on naming feature files with same name as the feature:
```suggestion
Feature: System Integration
```
Or you can name the file `system_integration_package_metrics.feature`

File: e2e/_suites/fleet/features/system_integration.feature
Comment (mdelapenya) [inline e2e/_suites/fleet/features/system_integration.feature:1]: To run this feature file in the tests you must add it to the `.ci/.e2e-tests.yml` descriptor, which defines the parallel branches to be spread in the CI workers and OS platforms (Please read https://github.com/elastic/e2e-testing/tree/master/e2e#feature-files-and-the-ci). I'd add this block to the YAML file (from L13 to L34), renaming the main annotation/tag for the feature file:

```yaml
  - suite: "fleet"
    platforms:
      - "arm64"
      - "ubuntu-18.04"
    scenarios:
      - name: "Fleet"
        pullRequestFilter: " && ~debian"
        tags: "fleet_mode_agent"
      - name: "Endpoint Integration"
        platforms:
          - "ubuntu-18.04"
        pullRequestFilter: " && ~debian"
        tags: "agent_endpoint_integration"
      - name: "Linux Integration"
        pullRequestFilter: " && ~debian"
        tags: "linux_integration"
      - name: "System Integration"
        pullRequestFilter: " && ~debian"
        tags: "system_integration"
      - name: "Stand-alone"
        pullRequestFilter: " && ~ubi8"
        tags: "stand_alone_agent"
      - name: "Backend Processes"
        pullRequestFilter: " && ~debian"
        tags: "backend_processes"
```

```suggestion
@system_integration
```

File: e2e/_suites/fleet/features/system_integration.feature
Comment (mdelapenya) [inline e2e/_suites/fleet/features/system_integration.feature:10]: I'd rename the step to remove the `we` part:
```suggestion
    And "system/metrics" with "core" metrics are present in the datastreams
```
Note that you should update all scenarios and the Go code that glues the step name with the method. Perfect if you also rename the method too

Comment (IgorGuz2000) [reply e2e/_suites/fleet/features/system_integration.feature]: Done

File: e2e/_suites/fleet/features/system_integration.feature
Comment (IgorGuz2000) [reply e2e/_suites/fleet/features/system_integration.feature:1]: Done

File: e2e/_suites/fleet/features/system_integration.feature
Comment (IgorGuz2000) [reply e2e/_suites/fleet/features/system_integration.feature]: Done

**Comment by mdelapenya** [comment]
This is great! Good job!!

The new scenario passed in AMD based installers but failed on ARM. I think we are excluding the ARM ones because the package integrations are not fully supported yet on ARM. I'm going to comment in the proper file how to skip that

File: .ci/.e2e-tests.yaml
Comment (mdelapenya) [inline .ci/.e2e-tests.yaml:29]: This will override the parent platforms element, as the item has precedence:
```suggestion
      - name: "System Integration"
        platforms:
          - "ubuntu-18.04"
```

Comment (IgorGuz2000) [reply .ci/.e2e-tests.yaml:29]: done

**Review by adam-stokes — CHANGES_REQUESTED** [review]
Great work! Can we remove those fmt.Println's from the code? Also, it may be better to put the `metrics.json` file in the same directory as the feature file to keep those things together. Thoughts?

**Comment by mdelapenya** [comment]
> Great work! Can we remove those fmt.Println's from the code? Also, it may be better to put the `metrics.json` file in the same directory as the feature file to keep those things together. Thoughts?

I'd create a `testresources` dir in the test suite, to separate a resource from the feature file

**Comment by IgorGuz2000** [comment]
Done all the changes

File: e2e/_suites/fleet/fleet.go
Comment (mdelapenya) [inline e2e/_suites/fleet/fleet.go:1189]: I meant moving the JSON file for tests to a `testresources` directory. Could you move the file there? 🙏 

```suggestion
	data := readJSONFile("./testresources/metrics.json", integration, set)
```

Comment (mdelapenya) [reply e2e/_suites/fleet/fleet.go:1189]: It would be great to also use `filepath` to honour current OS' path separator. There are several examples in the codebase for `filepath.Join`

Comment (adam-stokes) [reply e2e/_suites/fleet/fleet.go:1189]: We defintely need this to handle differences between operating systems

Comment (IgorGuz2000) [reply e2e/_suites/fleet/fleet.go]: Done

Comment (IgorGuz2000) [reply e2e/_suites/fleet/fleet.go]: Done

**Comment by IgorGuz2000** [comment]
Hi Adam What did you mean by /test

**Comment by adam-stokes** [comment]
> Hi Adam What did you mean by /test

That is a trigger that re-runs jenkins, I made a bunch of fixes to make our CI green again so I wanted to retest your PR against the latest code.

**Review by adam-stokes — APPROVED** [review]
This is an awesome addition, I love that we will be able to define our inputs via json rather than in the code directly

**Comment by adam-stokes** [comment]
I want to wait on @mdelapenya final approval before merge, but it all LGTM!

File: e2e/_suites/fleet/features/system_integration.feature
Comment (mdelapenya) [inline e2e/_suites/fleet/features/system_integration.feature:9]: I believe you can obtain the "<os>" from the current installer. If not, we should provide that cc/ @adam-stokes

Comment (adam-stokes) [reply e2e/_suites/fleet/features/system_integration.feature:9]: I think the formatting stripped what was between the quotes, @mdelapenya are you talking about querying the underlying OS via installer?

Comment (mdelapenya) [reply e2e/_suites/fleet/features/system_integration.feature:9]: The installer has a reference to the downloadable file, so at that point we can set the target OS: "centos" or "debian", which was set by the example outline when deploying an "<os>" agent with "tar" installer

Comment (mdelapenya) [reply e2e/_suites/fleet/features/system_integration.feature:9]: Thinking about it, I think we could revert this https://github.com/elastic/e2e-testing/pull/1291/commits/16fa4ca4296fad72f73e4b31fdfbcc477d412d60 commit and the PR would be great to go.

Could you do it 🙏 ? We can discuss about anything else about it in a follow-up issue, so that we can see this code merged the soonest.

Comment (IgorGuz2000) [reply e2e/_suites/fleet/features/system_integration.feature:9]: This was not  the reason I put the os in there

Comment (IgorGuz2000) [reply e2e/_suites/fleet/features/system_integration.feature:9]: It is because when you run with new command it is not clear what os you running

Comment (mdelapenya) [reply e2e/_suites/fleet/features/system_integration.feature:9]: Yep, I saw the logs in the commit to print the "os". For that, and because the "os" is already set in the first step `a "<os>" agent is deployed to Fleet with "tar" installer`, I commented with @adam-stokes about storing that value internally (maybe in the installer struct), so that it's possible to retrieve it later without having to make the steps that expressive.

OTOH, we'd like to hear your intention about the change, so that we can better understand and review it

Comment (mdelapenya) [reply e2e/_suites/fleet/features/system_integration.feature:9]: Another thing that comes to my mind is that, if we keep the step as is now: `the policy is updated to have "system/metrics" set to "core" for a "<os>" agent` it could be the case that an scenario writer picking valid steps could create an scenario in a wrong manner:

_Omitting example outlines for simplicity:_
```gherkin
Given a "centos" agent is deployed to Fleet with "tar" installer 
Then the policy is updated to have "system/metrics" set to "core" for a "unexpected-value-not-centos" agent
```

**Comment by IgorGuz2000** [comment]
Merging PR

**Pull request merged**
